drv/lpc55-syscon/src/main.rs

```diff
@@ -256,6 +267,10 @@ fn set_reset_reason() {
     };
 
     Jefe::from(JEFE.get_task_id()).set_reset_reason(reason);
+
+    // Need this to be able to use the syscon reset that works for CFPA
```

Comment:
Nit: doing this in `set_reset_reason` is confusing.  Maybe do it in `main` instead (moving ownership of `pmc`), e.g.
```rust
fn main() {
    // ... snip
    
    let pmc = unsafe { &*device::PMC::ptr() };
    set_reset_reason(pmc);

    // Need this to be able to use the syscon reset that works for CFPA
    // update
    pmc.resetctrl.write(|w| w.swrresetenable().enable());
    
    // ... etc
```

Reply:
I was writing a similar comment but was going to suggest putting it in `chip_reset()` :+1: